sql/src/main/java/io/crate/planner/operators/FetchOrEval.java

```diff
@@ -421,12 +423,21 @@ private static Symbol toInputColOrFetchRef(Symbol output,
                                                Map<FullQualifiedTableRelation, InputColumn> fetchInputColumnsByTable,
                                                BiConsumer<FullQualifiedTableRelation, Reference> allocateFetchRef,
                                                Map<Symbol, Symbol> expressionMapping,
-                                               @Nullable QualifiedName currentQualifiedName) {
+                                               @Nullable QualifiedName currentQualifiedName,
+                                               SubQueryResults subQueryResults,
+                                               Row params) {
         int idxInSource = sourceOutputs.indexOf(output);
         if (idxInSource > -1) {
             return new InputColumn(idxInSource, sourceOutputs.get(idxInSource).valueType());
         }
-        return FieldReplacer.replaceFields(output, f -> {
+
+        Symbol boundedOutput = SubQueryAndParamBinder.convert(output, params, subQueryResults);
```

Comment:
why don't we do this in the first place? like immediately converting the output?

Reply:
So first we check outputs without param binding and only then bind params and check again.
It is probably cheaper,  otherwise, i think, it shouldn't really make a difference..

Reply:
I don't think that this is worth the complexity, I'd prefer to bind first.